src/bootman/bootman.c

```diff
@@ -446,6 +457,208 @@ bool boot_manager_set_uname(BootManager *self, const char *uname)
         return self->have_sys_kernel;
 }
 
+static bool boot_manager_is_inird_valid_name(const char *name)
+{
+        if (strlen(name) < 3) {
+                return false;
+        }
+
+        return isdigit(name[0]) && isdigit(name[1]);
+}
+
+static int boot_manager_sort_initrd_name(const void *name1, const void *name2)
+{
+        return strcmp((char *)name1, (char *)name2);
+}
+
+bool boot_manager_get_initrd_freestanding(BootManager *self)
+{
+        autofree(DIR) *initrd_dir = NULL;
+        NcArray *dir_files = NULL;
+        struct dirent *ent = NULL;
+        struct stat st = { 0 };
+
+        if (!self || !self->initrd_freestanding_dir) {
+                return false;
+        }
+
+        dir_files = nc_array_new();
+        OOM_CHECK_RET(dir_files, false);
```

Comment:
So I'm a little confused by the use of OOM_CHECK_RET here since later on the DECLARE_OOM && abort() dance is used. Reasoning for returning if the allocation failes here?

Reply:
I already change the declare abort for OOM_CHECK_RET ro equivalent

Reply:
I guess I'm not sure why you would change it to return on memory allocation failure rather than aborting.

Reply:
ok, changing